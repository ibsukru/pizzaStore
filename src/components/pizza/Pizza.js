import React from 'react'

export default ({ sizes, onSelectingSize }) => {
  if (!onSelectingSize) return null
  if (!Array.isArray(sizes)) return null

  return (
    <section>
      <h2>Which size do you like?</h2>
      <div className='sizes'>
        {sizes.map(size => {
          return (
            <a
              className='sizes__size'
              onClick={() => onSelectingSize(size)}
              key={size.name}
            >
              {size.name}
            </a>
          )
        })}
      </div>
    </section>
  )
}
