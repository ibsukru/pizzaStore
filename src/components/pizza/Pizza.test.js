import React from 'react'
import renderer from 'react-test-renderer'
import Pizza from './Pizza'

[
  null,
  {
    onSelectingSize: null,
    sizes: []
  },
  {
    onSelectingSize: f => f,
    sizes: [
      {
        name: 'a size'
      }
    ]
  }
].forEach(props => {
  test(`With props=${JSON.stringify(props)}, should map snapshot`, () => {
    expect(renderer.create(<Pizza {...props} />).toJSON()).toMatchSnapshot()
  })
})
