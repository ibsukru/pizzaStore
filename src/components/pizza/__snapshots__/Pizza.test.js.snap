// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With props={"onSelectingSize":null,"sizes":[]}, should map snapshot 1`] = `null`;

exports[`With props={"sizes":[{"name":"a size"}]}, should map snapshot 1`] = `
<section>
  <h2>
    Which size do you like?
  </h2>
  <div
    className="sizes"
  >
    <a
      className="sizes__size"
      onClick={[Function]}
    >
      a size
    </a>
  </div>
</section>
`;

exports[`With props=null, should map snapshot 1`] = `null`;
